File: aula2/Personagem.java
public class Personagem{
    String nome;
    int vida;
    int forca;
    
    public void receberDano(int dano) {
        vida = vida - dano;
        System.out.println(nome + " sofreu " + dano + " de dano");
        if (vida<0){
            vida=0;
        }
    }
    public boolean estaVivo(){
        if (vida>0){
            return true;
        }
        else{
            return false;
        }
    }
    public void ficha(){
        System.out.println("\nNome: " + nome);
        System.out.println("Vida: " + vida);
        System.out.println("Força: " + forca);
    }
    public void atacar(Personagem alvo){
        System.out.println(nome + " ataca " + alvo.nome);
        alvo.receberDano(forca);
        
    }
}

public class Main{
    public static void main(String[] args){
        Personagem jogador = new Personagem();
        jogador.nome = "Raquel";
        jogador.vida = 100;
        jogador.forca = 10;
        
        Personagem chefe = new Personagem();
        chefe.nome = "Barbie";
        chefe.vida = 500;
        chefe.forca = 40;
        
        System.out.println("\n===Ficha===");
        jogador.ficha();
        chefe.ficha();
        System.out.println("\n===Batalha===");
        while(jogador.estaVivo() && chefe.estaVivo()){
            jogador.atacar(chefe);
            
            if (! chefe.estaVivo()){
                System.out.println(chefe.nome + " derrotado.");
                break;
            }
            chefe.atacar(jogador);

            if (! jogador.estaVivo()){
                System.out.println("\n" + jogador.nome + " derrotado!!!");
                break;
            }
        }
        System.out.println("\n===Ficha===");
        jogador.ficha();
        chefe.ficha();
    }
}

File: aula2/personagem.py
class personagem():
    def __init__(self, nome, vida, forca):
        self.nome = nome
        self.vida = vida
        self.forca = forca

    def receberDano(self, dano):
        self.vida = self.vida - dano
        print("%s recebeu %d de dano" %(self.nome, dano))
        if self.vida<0:
            self.vida=0
    
    def estaVivo(self):
        if self.vida>0:
            return True
        else:
            return False

    def ficha(self):
        print("\nNome: ", self.nome)
        print("Vida: ", self.vida)
        print("Força: ", self.forca)
    
    def atacar(self, alvo):
        print(self.nome, " ataca ", alvo.nome)
        alvo.receberDano(self.forca)

if __name__ == "__main__":

    jogador = personagem("Raquel", 100, 10)
    chefe = personagem("Barbie", 500, 40)

    print("===Ficha===")
    jogador.ficha()
    chefe.ficha()
    print("===Batalha===")
    while jogador.estaVivo() and chefe.estaVivo():
        jogador.atacar(chefe)
        if not chefe.estaVivo():
            break
        chefe.atacar(jogador)
        if not jogador.estaVivo():
            break

    print("===Ficha===")
    jogador.ficha()
    chefe.ficha()
